File: app/src/main/java/com/stathis/elmepaunivapp/ui/syllabus/recyclerview/SemesterAdapter.java
package com.stathis.elmepaunivapp.ui.syllabus.recyclerview;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.ListAdapter;

import com.stathis.elmepaunivapp.R;
import com.stathis.elmepaunivapp.abstraction.DiffItemCallbackClass;
import com.stathis.elmepaunivapp.listeners.activity_listeners.SyllabusActivityListener;
import com.stathis.elmepaunivapp.ui.syllabus.model.Semester;

public class SemesterAdapter extends ListAdapter<Semester, SemesterViewHolder> {

    private SyllabusActivityListener listener;

    public SemesterAdapter(SyllabusActivityListener listener) {
        super(new DiffItemCallbackClass<Semester>());
        this.listener = listener;
    }

    @NonNull
    @Override
    public SemesterViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new SemesterViewHolder(LayoutInflater.from(parent.getContext())
                .inflate(R.layout.holder_semester_item_row, parent, false),listener);
    }

    @Override
    public void onBindViewHolder(@NonNull SemesterViewHolder holder, int position) {
        holder.present(getItem(position));
    }
}

File: app/src/main/java/com/stathis/elmepaunivapp/ui/syllabus/recyclerview/SemesterAdapter.kt
package com.stathis.elmepaunivapp.ui.syllabus.recyclerview

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.ListAdapter
import com.stathis.elmepaunivapp.R
import com.stathis.elmepaunivapp.abstraction.DiffItemClass
import com.stathis.elmepaunivapp.callbacks.ElmepaClickListener
import com.stathis.elmepaunivapp.abstraction.LocalModel
import com.stathis.elmepaunivapp.databinding.HolderSemesterItemRowBinding
import com.stathis.elmepaunivapp.databinding.ProfessorItemRowBinding

class SemesterAdapter(private val callback : ElmepaClickListener) : ListAdapter<LocalModel, SemesterViewHolder>(DiffItemClass<LocalModel>()) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SemesterViewHolder {
        val view = HolderSemesterItemRowBinding.inflate(LayoutInflater.from(parent.context),parent,false)
        return SemesterViewHolder(view,callback)
    }

    override fun onBindViewHolder(holder: SemesterViewHolder, position: Int) {
        holder.bindData(getItem(position))
    }
}